launcher/src/main/java/com/skcraft/launcher/update/Updater.java

```diff
@@ -171,7 +168,7 @@ protected void update(Instance instance) throws Exception {
         // Download assets
         log.info("Enumerating assets to download...");
         progress = new DefaultProgress(-1, SharedLocale.tr("instanceUpdater.collectingAssets"));
-        installAssets(installer, version, launcher.propUrl("assetsIndexUrl", version.getAssetsIndex()), assetsSources);
+        installAssets(installer, version, url(version.getAssetIndex().get("url")), assetsSources);
```

Comment:
```suggestion
        URL assetUrl = version.getAssetIndex() != null
                ? url(version.getAssetIndex().get("url"))
                : launcher.propUrl("assetsIndexUrl", version.getAssetsIndex());
        installAssets(installer, version, assetUrl, assetsSources);
```

to support modpack created with old builder? Also assetsIndexUrl in the properties should not delete.

Reply:
Soon™ 

Mojang deprecate https://s3.amazonaws.com/Minecraft.Download/indexes/%s.json but I would prefer to remove this functionally from this launcher here if the link is really dead.

In mean time older mod pack are still working and need only rebuild if the S3 Endpoint is not available. This is a nice backward compatibility which is important, e.g. for @sk89q (see https://github.com/SKCraft/Launcher/pull/164#issuecomment-254714556)